src/applications/vaos/new-appointment/components/VAFacilityPage/VAFacilityPageV2.jsx

```diff
@@ -32,31 +41,184 @@ const pageKey = 'vaFacilityV2';
-  updateFormData,
-  typeOfCare,
-  routeToPreviousAppointmentPage,
   pageChangeInProgress,
+  parentDetails,
+  parentFacilities,
+  parentFacilitiesStatus,
+  routeToPreviousAppointmentPage,
+  routeToNextAppointmentPage,
+  schema,
+  singleValidVALocation,
+  siteId,
+  typeOfCare,
+  updateFormData,
 }) {
-  useEffect(() => {
-    document.title = `${pageTitle} | Veterans Affairs`;
-    scrollAndFocus();
-    openFacilityPageV2(pageKey, uiSchema, initialSchema);
-  }, []);
-
-  if (childFacilitiesStatus === FETCH_STATUS.loading) {
-    return <LoadingIndicator message="Finding locations" />;
+  const history = useHistory();
+  const loadingEligibility = loadingEligibilityStatus === FETCH_STATUS.loading;
+  const loadingParents = parentFacilitiesStatus === FETCH_STATUS.loading;
+  const loadingFacilities = childFacilitiesStatus === FETCH_STATUS.loading;
+  const [showEligibilityModal, setShowEligibilityModal] = useState(false);
+
+  useEffect(
+    () => {
+      document.title = `${pageTitle} | Veterans Affairs`;
+      scrollAndFocus();
+      openFacilityPageV2(pageKey, uiSchema, initialSchema);
+    },
+    [openFacilityPageV2],
+  );
+
+  useEffect(
+    () => {
+      if (!loadingEligibility && canScheduleAtChosenFacility === false) {
+        setShowEligibilityModal(true);
+      }
+    },
+    [loadingEligibility, canScheduleAtChosenFacility],
+  );
+
+  const goBack = () => routeToPreviousAppointmentPage(history, pageKey);
+
+  const goForward = () => routeToNextAppointmentPage(history, pageKey);
+
+  const onFacilityChange = newData => {
+    const selectedFacility = facilities?.find(f => f.id === newData.vaFacility);
+
+    const parentId = getParentOfLocation(parentFacilities, selectedFacility)
+      ?.id;
+
+    if (!!selectedFacility && !!parentId) {
+      updateFormData(pageKey, uiSchema, {
+        ...newData,
+        vaParent: parentId,
+      });
+      checkEligibility(selectedFacility, siteId);
+    }
+  };
+
+  const title = (
+    <h1 className="vads-u-font-size--h2">
+      Choose a VA location for your {typeOfCare} appointment
+    </h1>
+  );
+
+  if (hasDataFetchingError) {
+    return (
+      <div>
+        {title}
+        <ErrorMessage />
+      </div>
+    );
+  }
+
+  if (
+    loadingParents ||
+    loadingFacilities ||
+    (singleValidVALocation && loadingEligibility)
+  ) {
+    return (
+      <div>
+        {title}
+        <LoadingIndicator message="Finding locations" />
+      </div>
+    );
+  }
+
+  if (noValidVAParentFacilities) {
+    return (
+      <div>
+        {title}
+        <NoVASystems />
+        <div className="vads-u-margin-top--2">
+          <FormButtons
+            onBack={goBack}
+            disabled
+            pageChangeInProgress={pageChangeInProgress}
+            loadingText="Page change in progress"
+          />
+        </div>
+      </div>
+    );
+  }
+
+  if (noValidVAFacilities) {
+    return (
+      <div>
+        {title}
+        <NoValidVAFacilities
```

Comment:
We might need to tweak this content for the new page, since there can be more than one site, but no facilities.

Reply:
This will be covered in this new ticket here: https://app.zenhub.com/workspaces/vft-59c95ae5fda7577a9b3184f8/issues/department-of-veterans-affairs/va.gov-team/14152